### ISSUE TITLE
UI for workflows in Airy Inbox

### ISSUE THREAD

**Issue by lizhutter** [airyhq/airy#1250]
### Requirements

As previously discussed, the conversation that is currently selected should be highlighted in some way to make a distinction between conversations within the Inbox.

![Workflows - Inbox](https://user-images.githubusercontent.com/74183162/111504192-b55b0100-8747-11eb-883b-fee219a09b22.png)

In terms of states, "All, Open, Closed" buttons are seen in the image - however, system users should have the ability to create additional states if they desire. The user can click on these buttons to move the conversation from one state to another. It should also be clear which state the user is currently looking at; making the current state more prominent than the others.

It should also be possible to filter the conversations, as seen by the three horizontal lines in the top right corner of the Inbox column.

A system user should have the ability to expand the contact information column for the conversation that they are currently in. The system user should also have the ability to close this column with an "X" displayed in the contact information column.

In the contact information column, it should be possible to display the contact's name, local time, and email address, in addition to the possibility to connect their Facebook page, an external ID, phone number, and address. In this column, a system user has the ability to add tags to this contact/conversation.

The image below can be used for what happens when a system user clicks the filter:

![screenshot_2021-03-31_at_11 09 14](https://user-images.githubusercontent.com/74183162/113120940-6797cb80-9212-11eb-9806-5eaea8947812.png)

### Design Note

However, the above image should be used only as a guide; the design does not need to look like this.

**Comment by lucapette** [comment]
I will reply inline


> In terms of states, "All, Open, Closed" buttons are seen in the image - however, system users should have the ability to create additional states if they desire. The user can click on these buttons to move the conversation from one state to another.

As presented, there are two problems with the current design:

- it conflicts with the existing filtering section (there's already an "All/Unread       -  Filters" header in the conversations list
- there's very little space for more than two states (with very short names)

> 
> It should also be possible to filter the conversations, as seen by the three horizontal lines in the top right corner of the Inbox column.

- So "solves" one problem? I like the icon better than "Filters" for sure (and if we want to do that only we can extract to a "use icon for filters button". But I believe the two above-mentioned problems stay.

> 
> A system user should have the ability to expand the contact information column for the conversation that they are currently in. The system user should also have the ability to close this column with an "X" displayed in the contact information column.
> 

This seems unrelated but reasonable. I would extract it to a different issue so we can work on it right away

> In the contact information column, it should be possible to display the contact's name, local time, and email address, in addition to the possibility to connect their Facebook page, an external ID, phone number, and address. In this column, a system user has the ability to add tags to this contact/conversation.

two things:

- `it should be possible to display the contact's name, local time, and email address, in addition to the possibility to connect their Facebook page, an external ID, phone number, and address` we store none of this information, not we actually have a contact at the moment. So this I would discuss again in its own issue "introduce contacts" or something along the lines of that. As it's quite some conceptual work on the data model before we can even get to it

- We can't tag "contact/conversation". Right now the feature is conversation specific because contacts don't exist. Once we introduce the concept we can reevaluate what "tagging" means

**Comment by lizhutter** [comment]
Good points. We also need to think about these items for the UI as well:
- How to mark the status of a conversation: probably best as a button in the upper right side of the conversation. Say we mark a conversation as "closed" --> we need to be able to re-open it.
- When a system user clicks on the filter (three horizontal lines at the upper right side of the Inbox), what happens?
- Read/unread status: I don't think this is very important in this UI. I think that the All/Open/Closed status is sufficient (unless there's clear evidence proving this idea wrong).

**Comment by lucapette** [comment]
> Good points. We also need to think about these items for the UI as well:
> 
>     * How to mark the status of a conversation: probably best as a button in the upper right side of the conversation. Say we mark a conversation as "closed" --> we need to be able to re-open it.

I don't know what's a good UI solution for this. But it's indeed only a UI problem (as conceptually, it's easy to understand what this is supposed to do)

>     * When a system user clicks on the filter (three horizontal lines at the upper right side of the Inbox), what happens?

Good question :) I think it also depends on how the workflow filtering is going to look like? The proposed design doesn't scale well, so maybe the actual solution we come up with will make this question meaningless.

> 
>     * Read/unread status: I don't think this is very important in this UI. I think that the All/Open/Closed status is sufficient (unless there's clear evidence proving this idea wrong).

The UI looks broken without that as people are just used to messengers to make it clear one way or the other that a conversation is unread (we do not do anything at the moment.... and that's really not good enough). But I believe this is not connected to this issue and we may want to tackle it separately?

**Comment by lizhutter** [comment]
Closed with https://github.com/airyhq/airy/milestone/29
